sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Grpc/SettlementService.cs

```diff
@@ -74,16 +73,38 @@ public override async Task<Empty> Deadletter(DeadletterRequest request, ServerCa
             {
                 await tuple.Actions.DeadLetterMessageAsync(
                     tuple.Message,
-                    request.PropertiesToModify.ToDictionary(
-                        pair => pair.Key,
-                        pair => pair.Value.GetPropertyValue()),
+                    DeserializeAmqpMap(request.PropertiesToModify),
                     request.DeadletterReason,
                     request.DeadletterErrorDescription,
                     context.CancellationToken).ConfigureAwait(false);
                 return new Empty();
             }
             throw new RpcException (new Status(StatusCode.FailedPrecondition, $"LockToken {request.Locktoken} not found."));
         }
+
+        private static Dictionary<string, object> DeserializeAmqpMap(ByteString mapBytes)
+        {
+            if (mapBytes == null || mapBytes == ByteString.Empty)
+            {
+                return null;
+            }
+
+            var bytes = mapBytes.ToByteArray();
+            using ByteBuffer buffer = new ByteBuffer(bytes.Length, false);
+            AmqpBitConverter.WriteBytes(buffer, bytes, 0, bytes.Length);
```

Comment:
These are from the AMQP transport library?   Should we consider abstracting them in Core.Amqp or are they specific enough to this scenario that there's no value?